pkg/kn/commands/service/update_test.go

```diff
@@ -737,6 +737,15 @@ func TestServiceUpdateDeletionTimestampNotNil(t *testing.T) {
 	assert.ErrorContains(t, err, "service")
 }
 
+func TestServiceUpdateTagDoesNotExist(t *testing.T) {
+	orig := newEmptyService()
+
+	_, _, _, err := fakeServiceUpdate(orig, []string{
+		"service", "update", "foo", "--untag", "foo", "--no-wait"})
+
+	assert.Error(t, err, "Error: tag foo does not exist")
```

Comment:
I'd suggest to change this to (please add the import, too, if missing0:

```go
assert.Assert(t, util.ContainsAll(err.Error(), "tag", "foo", "not exist"))
```

It's always very fragile to assert on the exact wording of something, which might slight improvements tedious to work on (as then always a test needs to be fixed). Instead, just check that the important _context information_ is contained in the error message. This focus also having good error message which as much context information as possible to help a user to understand and fix the error. E.g. Here you could also add the service name which would help the user to understand which service you wanted to tag. The more context, the better.

Also, please use error message according to the golang standard: Starting lower case, no punctuation at the need. The "Error:" prefix is also redundant here (its an error anyway), and in fact the top-level routine which prints the error will add the error prefix anyway. So better remove it here, too.

We have an issues open (not sure where), where we want to check for all error messages to be consistently following this rule. Might be a good exercise to automate that and make it part of the build process.

Reply:
Just to clarify here, the `error: ` prefix won't be added unless included in the message it seems:

```
./kn service update hello --untag foo --no-wait

tag(s) foo not present for any revisions of service hello
```

Reply:
Totally agree, that a consistent error handling is missing, but for consistency's sake, I think there should be a single point which prepares an error message for print out.

In fact, in the refactoring in #877 there is a single exit point in case of a custom error, that now adds the appropriate prefix --> https://github.com/knative/client/blob/a71cc3fd91b9722e983929709c84ee9fd29939c5/cmd/kn/main.go#L38-L41

We can more error messaging mangling at this point (like uppercasing the first letter), but if we assume that the change above gets merged, then we should already avoid an `Error:` prefix.

Reply:
But don't let that block this PR as we have to go over all error messages later anyway.